### ISSUE TITLE
Display now activates but....

### ISSUE THREAD

**Issue by MacBandito** [fryc88/klipper-sv06plus-screen#8]
I must have missed a step or an update broke this process.  The display now turns on and shows the stock graphics with the OEM buttons and everything.  Before it would just sit at the Sovol startup screen.  As far as I can tell I went through all the steps you listed. I thought maybe I didn't flash the screen firmware correctly and tried again but this time it just shows 0 along all the files.  The first time it shows exactly as your image did on the instructions.   The only step that was different then your was that Mainsail never showed invalid for the firmware after I pulled the repository.   I may have added the -v -g arguments to Moonraker prior to loading  Mainsail though.  I don't recall.

![IMG_3281](https://github.com/fryc88/klipper-sv06plus-screen/assets/141544276/baf59683-1fc6-4e40-8627-12621d6787e1)

<img width="500" alt="Screenshot 2023-10-18 at 5 28 03 PM" src="https://github.com/fryc88/klipper-sv06plus-screen/assets/141544276/f5f08937-65ad-4823-bf8a-368125e1d34d">

**Comment by fryc88** [comment]
If your screen shows sovol logos that means it's not flashed. Try one more time with different card, preferably smaller capacity. Most of stock cards that come with printer work just fine, If i remember right that screen works with 8-16GB only.

With klipper branch it's very important to have gbkwiat as source because stock one won't work, but since you have gone through welcome screen on touchscreen then it looks like you have it there.

**Comment by MacBandito** [comment]
I did try that previously refreshing it again with the same card I use for flashing the main units firmware.  The first time I flashed it I got the exact same scree with all the same wording and numbers as shown in the instructions.  On refresh it shows zeros down the right side showing no files copied.  I just did it again with the original SD card just to be sure and I get the same result.

![IMG_3283](https://github.com/fryc88/klipper-sv06plus-screen/assets/141544276/aeecdb8d-3bc9-44e9-b936-10a6f66abbc8)

**Comment by fryc88** [comment]
So unfortunate that means it's still not being flashed correctly. try formatting card, using different one, etc. If it shows zeros on the right of each lines it means it didn't flash anything.

**Comment by fryc88** [comment]
If you see Sovols logo it literally means that screen never been flashed properly. That firmware from my github overwrites whole thing, so you should see Klipper logo instead. Until you see it, you can't consider it flashed.

**Comment by MacBandito** [comment]
Okay. Very weird.

**Comment by MacBandito** [comment]
Well I'm stumped.  I've tried different cards and I get all zeroes every time.  The only time it showed a count at the right was the first time I did this.  I tried the stock firmware for the screen same all zeroes.  I've tried reformatting, holding the card in there, shifting the card, plugging it in after it started, removing and reinserting while running, changing the names of the card, changing the names of all the files, and more.  It just gives me all zeroes now.  Guess I'll just by one of those klipper pads thought it would be fun to get this working.

**Comment by adrianjdebono** [comment]
Silly question but what sort of card did you use? I ended up using the sv06 plus 8gb card and it worked every time
I'd love to see if we can get it working @MacBandito . I did it to 10 of my sv06 plus with no issues

**Comment by frostius** [comment]
Can confirm the 8G card I got with the printer was able to flash.  I was not able to flash with a 8G partition on a 64G card.

**Comment by adrianjdebono** [comment]
@fryc88 hope you don't mind if it was possible to get a copy of the raw files to be able to edit the actual screen firmware in dgus? I had a few ideas to customise (I am actually using the firmware for a Voxelab Aries that is running a SV06 Plus motherboard, and I am using it in a 90 degree rotation hence i was hoping I could possibly borrow the files to edit.

**Comment by MacBandito** [comment]
> Silly question but what sort of card did you use? I ended up using the sv06 plus 8gb card and it worked every time I'd love to see if we can get it working @MacBandito . I did it to 10 of my sv06 plus with no issues

4 different brands.  The one that came with from Sovol, Sandisk, Samsung, PNY all 16gb or less.  The screen definitely acts like it’s loading the firmware but doesn’t actually load it.   If you want I can ship the screen to you to try.

**Comment by frostius** [comment]
@MacBandito My sovol 8G card is formatted as ex-fat, not fat32.  You might try formatting your card to ex-fat and see if that solves the problem.

**Comment by MacBandito** [comment]
I don’t believe I ever reformatted the sovol card so it should be the same.  I’ll double check however.  

I do want to reiterate that the display sees that the card is there and that there is a firmware file to load.  I know this because I’ve tried changing the name on the file and the firmware load screen does not come up unless it has the correct name.  So I don’t think the card is the issue.

**Comment by fryc88** [comment]
It is card issue definitely. Either damaged blocks or wrong filesystem.

**Comment by MacBandito** [comment]
I tried ex-fat but that was a no go.  I tried 2 cards.  The sovol Lexar 8GB the screen didn't load the firmware screen at all tried twice.  The screen would turn on and go to the Sovol splash screen and just stay there.  I then tried a 16gb SanDisk and the screen backlight would turn on but the screen would stay blank.  I reformatted the SanDisk back to Fat32 just to be sure and it again brought up the firmware load screen but didn't copy anything.

**Comment by MacBandito** [comment]
Anyone willing to send me a working SD card they've used with the firmware on it and I'll throw you $5-10 for your effort?

**Comment by MacBandito** [comment]
Good news everybody.   I found that the card I thought was the Sovol card was not and I had hid it away on my printer.  I used it and voila it works.  However now I have a new issue.  when I try to home using the screen it starts to home then immediately gives me a Shutdown screen.  Looks like Klipper is crashing.  I have to reboot clipper to get everything to come up.

**Comment by adrianjdebono** [comment]
That is awesome news. I am so glad it's all working again. I have a feeling it happened to me when using the home button on the screeb

**Comment by frostius** [comment]
Haha, my exact experience as well - spent hours getting the screen to work only for the home button to try to drive the nozzle thru my print bed.

I had hopes of a similar feature set to the sovol klipper screen.  The reality is pretty disappointing.  It doesn't support a retract/extract button during pause, no print start, no macros, the abort function doesn't actually work for me - I have to pause then abort.  

Fairly useless to be honest. I'll probably end up mounting a cheap tablet to the printer

**Comment by MacBandito** [comment]
Good to know. I’ll get the BTT screen then.

**Comment by to0w1r3d** [comment]
It didn't flash with the stock Sovol card, I switched to a 2GB/FAT32 card and it finally took.  Like OP, it always looked like it flashed with the CRC Check, but never did.  The only difference I noticed was the time it took to write the files...on the bad card it was immediate, but the good card took a few minutes.   Everything is working now - SV01 Pro, v0.12 Klipper.  Thanks!!

**Comment by rovangju** [comment]
Had a lot of problems, went thru many SD cards, none worked until...

Windows - Format FAT32 with block size of 4096. 
A card that had not worked consistently, all of a sudden: voila.

**Comment by samlii** [comment]
Just to add my observation here.  The card I got with my SV01 Pro was not flashing to the screen until I reformatted it as FAT32 with 4096 Blocks.  From what I've read this is the key, the block size.  @fryc88, there should probably be something added to the README about this.
